### PR TITLE
Oxidation remove UUID

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- Add `UUID` auto conversion for postgres
- Add compatibility test for postgres
- Remove python uuid call in binding

- [ ] Does this PR affect the User ?

### CODE REVIEW

**Review by Volham22 — CHANGES_REQUESTED** [review]

**Review by Volham22 — APPROVED** [review]

File: parsec/backend/swift_blockstore.py [hunk 1]
```diff
@@ -33,9 +33,8 @@ def side_effect(key: str) -> str:
 
 def build_swift_slug(organization_id: OrganizationID, id: BlockID) -> str:
     # The slug uses the UUID canonical textual representation (eg.
-    # `CoolOrg/3b917792-35ac-409f-9af1-fe6de8d2b905`) where `BlockID.__str__`
-    # uses the short textual representation (eg. `3b91779235ac409f9af1fe6de8d2b905`)
-    return f"{organization_id.str}/{id.uuid}"
+    # `CoolOrg/3b917792-35ac-409f-9af1-fe6de8d2b905`)
+    return f"{organization_id.str}/{id}"
```
Comment (FirelightFlagboy) [inline parsec/backend/swift_blockstore.py:37]: Given that comment, we may want to be more explicit and provide a `as_hyphenated` method

**Review by Volham22 — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

File: parsec/core/pki/accepter.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ async def accepter_list_submitted_from_backend(
 
     for enrollment in cast(list[dict[str, object]], rep["enrollments"]):
 
-        enrollment_id: UUID = cast(UUID, enrollment["enrollment_id"])
+        enrollment_id = cast(UUID, enrollment["enrollment_id"])
```
Comment (FirelightFlagboy) [inline parsec/core/pki/accepter.py:69]: Should use `EnrollmentID`

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: parsec/backend/postgresql/pki.py [hunk 3]
```diff
@@ -413,7 +413,7 @@ async def list(self, organization_id: OrganizationID) -> List[PkiEnrollmentListI
             )
             return [
                 PkiEnrollmentListItem(
-                    enrollment_id=entry["enrollment_id"],
+                    enrollment_id=UUID(entry["enrollment_id"]),
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply parsec/backend/postgresql/pki.py:416]: Done here: https://github.com/Scille/parsec-cloud/pull/3585

File: parsec/backend/postgresql/pki.py [hunk 4]
```diff
@@ -297,7 +297,7 @@ async def submit(
                         await conn.execute(
                             *_q_cancel_pki_enrollment(
                                 organization_id=organization_id.str,
-                                enrollment_id=row["enrollment_id"],
+                                enrollment_id=UUID(row["enrollment_id"]),
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply parsec/backend/postgresql/pki.py:300]: Done here: https://github.com/Scille/parsec-cloud/pull/3585

**Review by FirelightFlagboy — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Pull request merged**